src/react/components/common/tagInput/tagInput.tsx

```diff
@@ -463,19 +468,27 @@ export class TagInput extends React.Component<ITagInputProps, ITagInputState> {
             let deselect = selected && oldTagOperation === TagOperationMode.None;
 
             // Only fire click event if a region is selected
-            if (this.props.selectedRegions &&
-                this.props.selectedRegions.length > 0 &&
-                this.props.onTagClick) {
-                deselect = false;
-                this.props.onTagClick(tag);
+            const { selectedRegions, onTagClick } = this.props;
+            if (selectedRegions && selectedRegions.length && onTagClick) {
+                if ((this.isSelectionMark(selectedRegions[0].category) && this.isSelectionMark(tag.type))
+                    || (!this.isSelectionMark(selectedRegions[0].category) && !this.isSelectionMark(tag.type))) {
+                    onTagClick(tag);
```

Comment:
both this condition and the next condition execute the same block of code, Meaning there's no reason so separate the conditions?

Reply:
We're allow to add value if:
1. Both label and tag - are 'selectionMark';
2. Both are not 'selectionMarks;
3. If the tag is 'free' we allow any combinations

Reply:
So it should be one if statement: 
"Both label and tag - are 'selectionMark" || 
"Both are not 'selectionMarks" ||
"If the tag is 'free' we allow any combinations"

Not two, since they execute the same block of code

Reply:
Ok, make sense.

Reply:
ok, make sense